provider/azureProvider.js

```diff
@@ -178,7 +179,24 @@ return new BbPromise((resolve, reject) => {
       }
     }
 
-    const template = JSON.parse(fs.readFileSync(templateFilePath, 'utf8'));
+    let template = JSON.parse(fs.readFileSync(templateFilePath, 'utf8'));
+
```

Comment:
With the implementation you have, all the environment variables specified are injected into ARM template without an an option to explicitly choose. I would prefer customers providing any settings that need to be supplied to ARM template via parameters in yml file.
Can you try using variables:
- Specify AppSetitng in ARM template:  "[variables('KeyName')]": "[variables('EnvValue')]"
- Provide keyName in armTemplate variables
- If a value is not specified for the keyname, look up that keyName in serverless environment variables